ycmd/utils.py

```diff
@@ -266,6 +266,15 @@ def FindExecutable( executable ):
   return None
 
 
+def FindExecutableWithFallback( executable_path, fallback ):
+  if executable_path:
+    executable_path = FindExecutable( ExpandVariablesInPath( executable_path ) )
+    if not executable_path:
+      return None
```

Comment:
Shouldn't this return `executable_path` when `FindExecutable` actually returned something?

Reply:
Ah! This pull request is getting quite embarrassing.
Thanks for catching that. I pushed an update.

I should write a test for this thing, but I'm not sure how without mocking things like `os.path.isfile()` and if I mock everything, I'm not sure how useful the test would be.

@puremourning Thoughts?

@matteomaser Please use Reviewable.